### PR TITLE
Text editors, FTW

### PR DESCRIPTION
**Pull request by bcumming**

### CODE REVIEW

**Comment by bcumming** [comment]
cscs-ci run alps;system=todi;uarch=gh200;uenv=editors:24.7

**Comment by simonpintarelli** [comment]
`fd` would be nice to have too.

File: config.yaml [hunk 1]
```diff
@@ -52,8 +52,17 @@ uenvs:
...
       deploy:
         eiger: [zen2]
-        santis: [gh200]
         todi: [gh200]
+  editors:
+    "24.7":
+      recipes:
+        zen2: "24.7"
+        zen3: "24.7"
+        gh200: "24.7"
```
Comment (RMeli) [inline config.yaml:61]: What is the meaning of the version?

File: recipes/editors/24.7/config.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name: icon-wcp
```
Comment (RMeli) [inline recipes/editors/24.7/config.yaml:1]:
```suggestion
name: editors
```

File: recipes/editors/24.7/environments.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  - emacs +treesitter
+  - nano
+  - vim +python +lua
+  # additional tools that integrate well with tools
+  - lua
+  - node-js
+  - python@3.12
+  - rust
+  - ripgrep
+  - treesitter
```
Comment (RMeli) [inline recipes/editors/24.7/environments.yaml:19]:
```suggestion
  - tree-sitter
```

File: recipes/editors/24.7/environments.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  - neovim
+  - emacs +treesitter
+  - nano
+  - vim +python +lua
+  # additional tools that integrate well with tools
+  - lua
+  - node-js
+  - python@3.12
+  - rust
+  - ripgrep
```
Comment (RMeli) [inline recipes/editors/24.7/environments.yaml:18]:
```suggestion
  - ripgrep
  - fd
  - lazygit
```
`fd` requested by @simonpintarelli

File: config.yaml [hunk 5]
```diff
@@ -52,8 +52,17 @@ uenvs:
         gh200: v0.9/gh200
       deploy:
         eiger: [zen2]
-        santis: [gh200]
```
Comment (RMeli) [inline config.yaml:55]: This is a change related to `arbor`. Is it indended to have it here?

**Review by RMeli — CHANGES_REQUESTED** [review]

File: config.yaml (hunk 5 above)
Comment (bcumming) [reply config.yaml:55]: It was a drive-by fix

File: config.yaml (hunk 1 above)
Comment (bcumming) [reply config.yaml:61]: July 2024.
Putting the date that it was configured, for future reference.

File: recipes/editors/24.7/environments.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      - toolchain: gcc
+        spec: gcc
+  unify: true
+  mpi: null
+  specs:
+  # the editors
+  - neovim
+  - emacs +treesitter
+  - nano
+  - vim +python +lua
```
[the remark this answers is not in the document]
Comment (bcumming) [reply recipes/editors/24.7/environments.yaml:12]: I avoided adding `x`, because that requires building half the universe (i.e. everything x-windows).

**Comment by bcumming** [comment]
cscs-ci run alps;system=todi;uarch=gh200;uenv=editors:24.7

File: config.yaml (hunk 1 above)
Comment (RMeli) [reply config.yaml:61]: I did not think of that... =)

**Comment by bcumming** [comment]
cscs-ci run alps;system=todi;uarch=gh200;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=todi;uarch=gh200;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=todi;uarch=gh200;uenv=editors:24.7

File: recipes/editors/24.7/environments.yaml (hunk 4 above)
Comment (bcumming) [reply recipes/editors/24.7/environments.yaml:21]: added

**Comment by bcumming** [comment]
cscs-ci run alps;system=todi;uarch=gh200;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=eiger;uarch=zen2;uenv=editors:24.7

**Comment by RMeli** [comment]
Would it be possible to add `direnv` too? Thanks.

**Comment by bcumming** [comment]
cscs-ci run alps;system=eiger;uarch=zen2;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=eiger;uarch=zen2;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=eiger;uarch=zen2;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=eiger;uarch=zen2;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=eiger;uarch=zen2;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=eiger;uarch=zen2;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=daint;uarch=gh200;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=eiger;uarch=zen2;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=daint;uarch=gh200;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=daint;uarch=gh200;uenv=arbor:v0.9

**Comment by bcumming** [comment]
cscs-ci run alps;system=eiger;uarch=zen2;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=daint;uarch=gh200;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=todi;uarch=gh200;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=eiger;uarch=zen2;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=daint;uarch=gh200;uenv=editors:24.7

**Comment by bcumming** [comment]
cscs-ci run alps;system=todi;uarch=gh200;uenv=editors:24.7

**Pull request merged**